packages/@ember/-internals/views/lib/system/event_dispatcher.js

```diff
@@ -335,7 +336,9 @@ export default EmberObject.extend({
           while (
             target &&
             target.nodeType === 1 &&
-            (!related || (related !== target && !target.contains(related)))
+            (!related ||
```

Comment:
I know this isn't new, but I'm curious why are we checking `!related` here? Can we tweak this to avoid boolean coercion? What do we expect `target.relatedTarget` to be? I was thinking it could only be either `undefined` or `null`, if thats correct we could (possibly in a different PR?) tweak to `related === undefined && related === null`...

Reply:
According to the [specs](https://www.w3.org/TR/uievents/#dom-mouseevent-relatedtarget) the  uninitialized value must be `null`, so I changes this to a strict null check.